docs/hardhat-network/reference/README.md

```diff
@@ -0,0 +1,407 @@
+# Hardhat Network Reference
+
+## Supported hardforks
+
+- byzantium
+- constantinople
+- petersburg
+- istanbul
+- muirGlacier
+
+## Config
+
+- Default behavior: Hardhat will automatically run in-process when `defaultNetwork` is empty or set to `hardhat`.
+
+### Mining modes
+
+<!-- TODO: add a link to the explanation of mining modes -->
+
+You can configure the mining behavior under your Hardhat Network settings:
+
+```js
+networks: {
+  hardhat: {
+    mining: {
+      auto: false,
+      interval: 5000
+    }
+  }
+}
+```
+
+In this example, automining is disabled and interval mining is set so that a new block is generated every 5 seconds. You can also configure interval mining to generate a new block after a random delay:
+
+```js
+networks: {
+  hardhat: {
+    mining: {
+      auto: false,
+      interval: [3000, 6000]
+    }
+  }
+}
+```
+
+In this case, a new block will be mined after a random delay of between 3 and 6 seconds. For example, the first block could be mined after 4 seconds, the second block 5.5 seconds after that, and so on.
+
+#### Manual mining
+
+You can disable both mining modes like this:
+
+```js
+networks: {
+  hardhat: {
+    mining: {
+      auto: false,
+      interval: 0
+    }
+  }
+}
+```
+
+This means that no new blocks will be mined by the Hardhat Network, but you can manually mine new blocks using the `evm_mine` RPC method. This will generate a new block that will include as many pending transactions as possible.
+
+### Mempool behavior
+
+When automine is disabled, every sent transaction is added to the mempool, which contains all the transactions that could be mined in the future. Hardhat Network's mempool follows the same rules as geth. This means, among other things, that:
+
+- Transactions with a higher gas price are included first
+- If two transactions can be included and both have the same gas price, the one that was received first is included first
```

Comment:
See: https://github.com/nomiclabs/hardhat/pull/1657/files#r674852709

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/20aa02e0a31da82939a609c8ccb57c6d38b9401d